asapdiscovery-data/asapdiscovery/data/schema_v2/target.py

```diff
@@ -0,0 +1,110 @@
+    )
+
+    fragalysis_id: Optional[str] = Field(
+        None, description="The PDB code of the target if applicable"
+    )
+
+    pdb_code: Optional[str] = Field(
+        None, description="The PDB code of the target if applicable"
+    )
+
+
+class Target(DataModelAbstractBase):
+    """
+    Schema for a Target
+    """
+
+    target_name: str = Field(None, description="The name of the target")
+
+    ids: Optional[TargetIdentifiers] = Field(
+        None,
+        description="TargetIdentifiers Schema for identifiers associated with this ligand",
+    )
+
+    data: str = Field(
+        "",
+        description="PDB file stored as a string to hold internal data state",
+        repr=False,
+    )
+    data_format: DataStorageType = Field(
+        DataStorageType.pdb,
+        description="Enum describing the data storage method",
+        allow_mutation=False,
+    )
+
+    @classmethod
+    def from_pdb(
+        cls, pdb_file: Union[str, Path], target_name: Optional[str] = None, **kwargs
+    ) -> "Target":
+        # directly read in data
+        pdb_str = read_file_directly(pdb_file)
+        return cls(data=pdb_str, target_name=target_name, **kwargs)
+
+    @classmethod
+    def from_pdb_via_openeye(
+        cls, pdb_file: Union[str, Path], target_name: Optional[str] = None, **kwargs
+    ) -> "Target":
+        # directly read in data
+        pdb_str = read_file_directly(pdb_file)
+        # NOTE: tradeof between speed and consistency with `from_pdb` method lines below will make sure that the pdb string is
+        # consistent between a load and dump by roundtripping through and openeye mol but will slow down the process significantly.
+        mol = pdb_string_to_oemol(pdb_str)
+        pdb_str = oemol_to_pdb_string(mol)
+        return cls(data=pdb_str, target_name=target_name, **kwargs)
+
+    def to_pdb(self, filename: Union[str, Path]) -> None:
+        # directly write out data
+        write_file_directly(filename, self.data)
+
+    @classmethod
+    def from_oemol(
+        cls, mol: oechem.OEMol, target_name: Optional[str] = None, **kwargs
+    ) -> "Target":
+        pdb_str = oemol_to_pdb_string(mol)
+        return cls(data=pdb_str, target_name=target_name, **kwargs)
+
+    def to_oemol(self) -> oechem.OEMol:
+        return pdb_string_to_oemol(self.data)
+
+    """
+    we are deferring responsibility of writing to and from OEDesignUnit to the caller
+    as it doesn't really make sense to force a specific OESpruce workflow on the user.
+    as there are so many different ways to generate a OEDesignUnit from a PDB file
+    depending on the options used.
+
+    Therefore the user is responsible for reading and outputting an OEMol or OEGraphMol of the OEDesignUnit
+    component in question.
+
+    eg.
+
+    L = Ligand.from_pdb('complex.pdb')
+    prepped_oemol = prep_oemol(L.to_oemol())
+    L2 = Ligand.from_oemol(prepped_oemol)
+    """
```

Comment:
I think we would like to be able to pass a DU around along with this schema, whatever that takes. As mentioned its pretty nontrivial to go DU --> Mol --> DU so if we can pass a DU directly (even if that means some godawful binary string) I think that would be preferabl

Reply:
I'm going to investigate DUs seperately as part of #355